mud-contracts/smart-object-framework-v2/src/namespaces/evefrontier/systems/entity-system/EntitySystem.sol

```diff
@@ -5,48 +5,52 @@ import { IWorldKernel } from "@latticexyz/world/src/IWorldKernel.sol";
 import { RESOURCE_SYSTEM } from "@latticexyz/world/src/worldResourceTypes.sol";
 
-import { Classes, ClassesData } from "../codegen/tables/Classes.sol";
-import { ClassSystemTagMap } from "../codegen/tables/ClassSystemTagMap.sol";
-import { ClassObjectMap, ClassObjectMapData } from "../codegen/tables/ClassObjectMap.sol";
-import { Objects } from "../codegen/tables/Objects.sol";
-import { IWorld } from "../codegen/world/IWorld.sol";
+import { Classes, ClassesData } from "../../codegen/tables/Classes.sol";
+import { ClassSystemTagMap } from "../../codegen/tables/ClassSystemTagMap.sol";
+import { ClassObjectMap, ClassObjectMapData } from "../../codegen/tables/ClassObjectMap.sol";
+import { Objects } from "../../codegen/tables/Objects.sol";
 
-import { Id, IdLib } from "../libs/Id.sol";
-import { ENTITY_CLASS, ENTITY_OBJECT } from "../types/entityTypes.sol";
-import { TAG_SYSTEM } from "../types/tagTypes.sol";
+import { Id, IdLib } from "../../../../libs/Id.sol";
+import { ENTITY_CLASS, ENTITY_OBJECT } from "../../../../types/entityTypes.sol";
+import { TAG_SYSTEM } from "../../../../types/tagTypes.sol";
 
-import { ITags } from "../interfaces/ITags.sol";
-import { IErrors } from "../interfaces/IErrors.sol";
+import { ITagSystem } from "../../interfaces/ITagSystem.sol";
+import { IEntitySystem } from "../../interfaces/IEntitySystem.sol";
 
-import { SmartObjectSystem } from "./inherit/SmartObjectSystem.sol";
+import { Utils as TagSystemUtils } from "../tag-system/Utils.sol";
+
+import { SmartObjectFramework } from "../../../../inherit/SmartObjectFramework.sol";
 
 /**
- * @title Entities
+ * @title EntitySystem
  * @author CCP Games
  * @dev Manage Class and Object creation/deletion through the use of reference Ids { see, `libs/Id.sol` and `types/entityTypes.sol`}
  */
-contract Entities is SmartObjectSystem {
+contract EntitySystem is IEntitySystem, SmartObjectFramework {
   /**
    * @notice register a Class Entity into the SOF with an initial set of assigned SystemTags
    * @param classId A unique ENTITY_CLASS type Id for referencing a newly registred Class Entity within SOF compatible Systems
    * @param systemTags An array of TAG_SYSTEM type Ids which correlate to exsiting MUD System ResourceIds for tagging a Class with
    */
   function registerClass(Id classId, Id[] memory systemTags) public {
     if (Id.unwrap(classId) == bytes32(0)) {
-      revert IErrors.InvalidEntityId(classId);
+      revert InvalidEntityId(classId);
     }
     if (classId.getType() != ENTITY_CLASS) {
       bytes2[] memory expected = new bytes2[](1);
       expected[0] = ENTITY_CLASS;
-      revert IErrors.WrongEntityType(classId.getType(), expected);
+      revert WrongEntityType(classId.getType(), expected);
     }
     if (Classes.getExists(classId)) {
-      revert IErrors.ClassAlreadyExists(classId);
+      revert ClassAlreadyExists(classId);
     }
 
     Classes.set(classId, true, new bytes32[](0), new bytes32[](0));
```

Comment:
why is the systemTags set to `new bytes32[](0)` ? isn't it supposed to include the systemTags ?